### PR TITLE
Restack, add various fixes branch

### PR DESCRIPTION
**Pull request by MasterPtato**

### CODE REVIEW

**Comment by MasterPtato** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/373?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #324** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/324?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #314** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/314?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #310** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/310?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #311** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/311?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #312** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/312?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #313** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/313?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #315** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/315?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #316** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/316?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #326** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/326?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #325** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/325?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #329** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/329?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                      * **PR #330** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/330?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                        * **PR #331** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/331?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                          * **PR #332** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/332?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                            * **PR #333** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/333?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                              * **PR #334** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/334?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                * **PR #335** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/335?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                  * **PR #336** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/336?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                    * **PR #337** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/337?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                      * **PR #338** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/338?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                        * **PR #339** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/339?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                          * **PR #340** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/340?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                            * **PR #341** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/341?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                              * **PR #342** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/342?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                * **PR #343** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/343?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                  * **PR #344** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/344?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                    * **PR #345** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/345?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                      * **PR #346** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/346?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                        * **PR #347** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/347?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                          * **PR #348** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/348?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                            * **PR #349** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/349?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                              * **PR #350** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/350?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                                * **PR #351** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/351?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                                  * **PR #352** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/352?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                                    * **PR #353** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/353?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                                      * **PR #354** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/354?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                                        * **PR #355** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/355?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                                          * **PR #356** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/356?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                                            * **PR #361** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/361?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                                              * **PR #362** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/362?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                                                * **PR #363** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/363?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                                                  * **PR #364** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/364?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                                                    * **PR #367** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/367?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                                                      * **PR #368** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/368?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                                                        * **PR #369** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/369?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                                                          * **PR #370** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/370?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                                                            * **PR #373** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/373?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rivet-gg/rivet/373?utm_source=stack-comment).

File: svc/pkg/faker/ops/job-template/templates/allocate.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	let args: Vec<String> = env::args().collect();
+	if args.len() < 2 {
+		eprintln!("missing size arg");
+		std::process::exit(1);
+	}
+	let size = args[1].parse::<usize>().expect("Error parsing size");
+
+	// Allocate the array
+	println!("allocating {size} bytes");
+	let _array = vec![0u8; size];
```
[the remark this answers is not in the document]
Comment (AngelOnFira) [reply svc/pkg/faker/ops/job-template/templates/allocate.rs:16]: Oop, looks like the compiler still rolls with it 👍

**Review by AngelOnFira — APPROVED** [review]

**Review by AngelOnFira — CHANGES_REQUESTED** [review]
Oops meant to ask for fixes on the change to the Dockerfile

File: lib/job-runner/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 FROM clux/muslrust:1.73.0
 
 WORKDIR /app
-COPY Cargo.toml Cargo.lock .
+COPY Cargo.toml Cargo.loc[k] .
```
[the remark this answers is not in the document]
Comment (MasterPtato) [reply lib/job-runner/Dockerfile:4]: Thats a good point. Removing it from the gitignore is probably the right play

**Review by AngelOnFira — APPROVED** [review]

File: lib/bolt/core/src/dep/cargo/cli.rs [hunk 3]
```diff
@@ -148,8 +148,14 @@ pub async fn build<'a, T: AsRef<str>>(ctx: &ProjectContext, opts: BuildOpts<'a,
 
 						FROM rust:1.72-slim
 
+						RUN apt-get update && apt-get install -y protobuf-compiler pkg-config libssl-dev g++
 						RUN apt-get update && apt-get install -y protobuf-compiler pkg-config libssl-dev g++
```
Comment (AngelOnFira) [inline lib/bolt/core/src/dep/cargo/cli.rs:152]: item dupe

File: lib/bolt/core/src/dep/cargo/cli.rs [hunk 4]
```diff
@@ -148,8 +148,14 @@ pub async fn build<'a, T: AsRef<str>>(ctx: &ProjectContext, opts: BuildOpts<'a,
...
 
+						RUN apt-get install --yes libpq-dev wget
+						RUN wget https://github.com/mozilla/sccache/releases/download/v0.2.15/sccache-v0.2.15-x86_64-unknown-linux-musl.tar.gz \
+							&& tar xzf sccache-v0.2.15-x86_64-unknown-linux-musl.tar.gz \
+							&& mv sccache-v0.2.15-x86_64-unknown-linux-musl/sccache /usr/local/bin/sccache \
+							&& chmod +x /usr/local/bin/sccache
 						RUN apt-get install --yes libpq-dev wget
 						RUN wget https://github.com/mozilla/sccache/releases/download/v0.2.15/sccache-v0.2.15-x86_64-unknown-linux-musl.tar.gz \
 							&& tar xzf sccache-v0.2.15-x86_64-unknown-linux-musl.tar.gz \
```
Comment (AngelOnFira) [inline lib/bolt/core/src/dep/cargo/cli.rs:161]: <i class='graphite__hidden'>[Re: lines 159 to 163]<br/><br/></i>

double dupe<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/373?utm_source=unchanged-line-comment">Graphite</a>.</span>

**Review by AngelOnFira — CHANGES_REQUESTED** [review]

**Review by AngelOnFira — APPROVED** [review]

**Review by NathanFlurry — APPROVED** [review]

**Pull request merged**